### DISCUSSION TITLE
Funktion des setvalue  "ww_pumpen_zyklus"

### DISCUSSION THREAD

**Discussion by decanyon** [dewenni/ESP_Buderus_KM271#90] [Q&A]
Hallo, hat jemand den Setvalue für die Zirkulationspumpe über MQTT getestet? 
 Gruß Dennis

**Comment by dewenni** [comment]
ja, habe es gerade nochmal getestet. Sollte funktionieren.
Wenn du die mqtt Sprache auf deutsch eingestellt hast, dann: `../setvalue/ww_pumpen_zyklus`

Falls nicht, wäre die erste Überprüfung ob der Befehl angenommen wird:

<img width="592" alt="image" src="https://github.com/dewenni/ESP_Buderus_KM271/assets/46074831/5403a3e0-810a-4c77-b7e9-11df6fe1213a">

**Reply by decanyon** [reply]
Der MQTT Befehl wird scheinbar nicht angenommen. Payload zwischen `0 und 7` bringt keinen Erfolg. Über die WebUI funktioniert die Einstellung der Pumpe, entsprechend kommt auch die Message auf dem MQTT Explorer.
![image](https://github.com/dewenni/ESP_Buderus_KM271/assets/128498986/02f5c059-c54a-4acf-a586-b4afe3a35625)

**Reply by dewenni** [reply]
so müsste das z.B. funktionieren:

<img width="396" alt="image" src="https://github.com/dewenni/ESP_Buderus_KM271/assets/46074831/4bb18f0d-2119-46bc-9ae1-06546c8dc3c6">

**Reply by decanyon** [reply]
Leider bringt das keinen Erfolg. Die "publish" Nachrichten funktionieren einwandfrei.

**Reply by dewenni** [reply]
ich habe tatsächlich noch eine mögliche Fehlerursache gefunden.
In der bisherigen Version gab es nur eine Einstellung für die Sprache. Diese war dann für das Web-Interface und gleichzeitig für die mqtt-Nachrichten.
Mit der V4 habe ich das jetzt aufgetrennt und eine separate Spracheinstellung für die mqtt-Nachtichten eingeführt. (config.mqtt.lang)

Ich habe aber im Code bei der Auswertung der mqtt-Befehle vergessen das anzupassen. Da schaue ich immer noch auf den alten config Wert für die Sprache (config.lang), der jetzt aber nur noch für das Web-Interface genutzt wird.

Wenn du also jetzt für die Weboberfläche eine andere Sprache eingestellt hast wie für die mqtt Nachrichten, dann kann das ein Problem sein. Das habe ich leider übersehen und werde ich spätestens am Wochenende fixen.

**Reply by decanyon** [reply]
Das Setzen über Mqtt funktioniert in meinem Fall mit beiden Sprache DE und EN nicht. 

Zu deiner Beschreibung nehme ich an, dass die Sprache dann für die Topics _undefined_ ist und daher generell keine Setvalues empfangen werden, da wir eine Abhängigkeit zum Sprachparameter haben.

**Reply by decanyon** [reply]
Bis das mit der Sprach config gelöst ist, kann man einmal beide Sprachparameter auf der WebUI auswählen. Dann ist der Wert gesetzt und die Mqtt Setpoints können über `.../setvalues/...` gesetzt werden .

**Reply by dewenni** [reply]
Genau, es sollte generell schon funktionieren.
Aktuell ist die Sprache für die Kommandos abhängig von der WebUI Sprache und das was der ESP sendet, also die ganzen Status und Config Werte sind abhängig der Sprache die unter den mqtt Einstellung vorgegeben ist.

Funktioniert das jetzt unter den Bedingungen bei dir?

Das Update mache ich heute Abend oder morgen fertig.

**Reply by dewenni** [reply]
=> [Release v4.0.3](https://github.com/dewenni/ESP_Buderus_KM271/releases/tag/v4.0.3)

**Reply by decanyon** [reply]
Funktioniert jetzt tadellos. Bin an der Integration in den HomeAssistant dran und bisher ist das mqtt feature recht zuverlässig. Vielen Dank für die kurzen Reaktionszeiten!  Das Update mache ich nach der Integration...

**Reply by dewenni** [reply]
Ich habe auch schon überlegt ob ich noch das [mqtt Discovery](https://www.home-assistant.io/integrations/mqtt/#mqtt-discovery) mit einbaue. Dann würden die Daten auch automatisch in HA über mqtt eingebunden werden. Das funktioniert auch recht gut. In dem Fall werden die einzelnen Daten auch einem Device zugeordnet. Ich hatte das vor ein paar Tagen mal mit ein paar Werten getestet. Sollte dann auch für OpenHAB funktionieren.
<img width="672" alt="image" src="https://github.com/dewenni/ESP_Buderus_KM271/assets/46074831/74a2a9af-b84d-40f5-909c-8e19013f35d0">

andererseits kann man für HA auch die Software von Daniel nutzen. Ich will ihm da auch keine Konkurrent machen :-)

**Reply by decanyon** [reply]
> => [Release v4.0.3](https://github.com/dewenni/ESP_Buderus_KM271/releases/tag/v4.0.3)

Ich vermute mehrere Bugs in der neuen Version. Hoffe es hilft weiter...

In dieser Version kann die Zeit nicht manuell gesetzt werden. Hab den NTP Sync nicht auch hinbekommen. 

Der Neustart scheint Probleme zu machen. Mehrere Anläufe waren nötig, bis sich das Modul mit dem WiFi verbunden hat.

**Reply by the78mole** [reply]
Hallo @dewenni ,

das mit dem Discovery und HA funktioniert initial ganz gut, allerdings hatte ich einige Probleme damit, dass HA sich das merkt. Bis gerade eben habe ich die Discovery-Message alle 15 Minuten ohne Retain-Flag geschickt. Warum ich das vor ein paar Monaten rausgenommen hatte, weiß ich nicht mehr, aber irgend einen Grund hatte es bestimmt. Ich aktiviere es jetzt mal wieder, verlängere das resend auf 12 Stunden und schaue, was passiert.

**Reply by Eule01x** [reply]
Ich fänd das spitze, wenn sich die Heizung einfach per Autodiscovery in Homeassistant mit all ihren Werten melden würde.

**Reply by dewenni** [reply]
@Eule01x ich habe da mal was vorbereitet :-)
Teste gerne mal das neue Release und gib mir Feedback. Dann am besten aber ein einer neuen Diskussion bezogen auf das Discovery.
